Formula/mingw-w64.rb

```diff
@@ -0,0 +1,220 @@
+  sha256 "9bb5cd7df78817377841a63555e73596dc0af4acbb71b09bd48de7cf24aeadd2"
+
+  option "without-i686", "Compile without i686 compiler"
+  option "without-x86_64", "Compile without x86_64 compiler"
+  # option "without-multilib", "Compile x86_64 copmiler without multilib support"
+
+  depends_on "texinfo" => :build
+  depends_on "mpfr"
+  depends_on "libmpc"
+  depends_on "isl"
+  depends_on "gmp"
+  depends_on "wine" => :optional
+
+  resource "binutils" do
+    url "https://ftpmirror.gnu.org/binutils/binutils-2.27.tar.gz"
+    sha256 "26253bf0f360ceeba1d9ab6965c57c6a48a01a8343382130d1ed47c468a3094f"
+  end
+
+  resource "gcc" do
+    url "https://ftpmirror.gnu.org/gcc/gcc-6.3.0/gcc-6.3.0.tar.bz2"
+    sha256 "f06ae7f3f790fbf0f018f6d40e844451e6bc3b7bc96e128e63b09825c1f8b29f"
+  end
+
+  def target_archs
+    if build.without?("i686") && build.without?("x86_64")
+      odie "Options --without-i686 and --without-x86_64 are mutually exclusive"
+    end
+    archs = []
+    archs << "i686-w64-mingw32" if build.with?("i686")
+    archs << "x86_64-w64-mingw32" if build.with?("x86_64")
+  end
+
+  def install
+    target_archs.each do |target_arch|
+      resource("binutils").stage do
+        args = %W[
+          --target=#{target_arch}
+          --disable-werror
+          --prefix=#{prefix}
+          --with-sysroot=#{prefix}
+          --infodir=#{info}
+          --mandir=#{man}
+        ]
+        if target_arch.start_with?("i686", "x86_64")
+          args << "--enable-targets=#{target_arch}" << "--disable-multilib"
+        end
+        # elsif target_arch.start_with?("x86_64") and build.with?("multilib")
+        #  args << "--enable-targets=x86_64-w64-mingw32,i686-w64-mingw32" << "--enable-multilib"
+        mkdir "build-#{target_arch}" do
+          system "../configure", *args
+          system "make"
+          system "make", "install-strip"
+        end
+      end
+      ENV.prepend_path "PATH", bin.to_s
+
+      args = %W[
+        --host=#{target_arch}
+        --prefix=#{prefix}/#{target_arch}
+        --enable-secure-api
+      ]
+      args << "--disable-multilib" if target_arch.start_with?("i686", "x86_64")
+      # elsif target_arch.start_with?("x86_64") and build.with?("multilib")
+      #  args << "--enable-sdk=all"
+
+      chdir "mingw-w64-headers" do
+        mkdir "build-#{target_arch}" do
+          system "../configure", *args
+          system "make"
+          system "make", "install"
+        end
+      end
+
+      ln_s "#{prefix}/#{target_arch}", "#{prefix}/mingw"
+      ln_s "#{prefix}/#{target_arch}/lib", "#{prefix}/#{target_arch}/lib64"
+
+      resource("gcc").stage buildpath/"gcc"
+
+      gcc_args = %W[
+        --target=#{target_arch}
+        --prefix=#{prefix}
+        --with-sysroot=#{prefix}
+        --disable-nls
+        --with-system-zlib
+        --enable-version-specific-runtime-libs
+        --enable-libstdcxx-time=yes
+        --enable-stage1-checking
+        --enable-checking=release
```

Comment:
`--enable-stage1-checking` is useless on a cross-compiler (i.e. non bootstrapped) build.
`--enable-checking=release`, `--enable-lto`, `--enable-libssp`, and `--enable-shared` are on by default, so we can remove them.

Regarding `--enable-libcc1`, I am not sure how useful it is for a cross-compiler, but it should also be the default if supported.